js/derivadex.js

```diff
@@ -0,0 +1,599 @@
+                'networksById': {
+                    'ETH': 'ERC20',
+                },
+            },
+            'fees': {
+                'trading': {
+                    'feeSide': 'get',
+                    'tierBased': false,
+                    'percentage': true,
+                    'taker': 0.002,
+                    'maker': 0.0,
+                },
+            },
+        });
+    }
+
+    async fetchMarkets (params = {}) {
+        /**
+         * @method
+         * @name derivadex#fetchMarkets
+         * @description retrieves data on all markets for derivadex
+         * @param {object} params extra parameters specific to the derivadex api endpoint
+         * @returns {[object]} an array of objects representing market data
+         */
+        params['kind'] = 0;
+        const response = await this.publicGetSpecs (params);
+        // {
+        //     "value": [
+        //         {
+        //             "kind": 0,
+        //             "name": "DDXPERP",
+        //             "expr": "\n(Market :name "DDXPERP"\n :tick-size 0.1\n :max-order-notional 0\n :max-taker-price-deviation 0.02\n :min-order-size 0.0001\n)",
+        //             "value": {
+        //                 "tickSize": "0.1",
+        //                 "minOrderSize": "0.0001",
+        //                 "maxOrderNotional": "0",
+        //                 "maxTakerPriceDeviation": "0.02"
+        //             }
+        //         },
+        //         {
+        //             "kind": 0,
+        //             "name": "BTCPERP",
+        //             "expr": "\n(Market :name "BTCPERP"\n :tick-size 1\n :max-order-notional 1000000\n :max-taker-price-deviation 0.02\n :min-order-size 0.00001\n)",
+        //             "value": {
+        //                 "tickSize": "1",
+        //                 "minOrderSize": "0.00001",
+        //                 "maxOrderNotional": "1000000",
+        //                 "maxTakerPriceDeviation": "0.02"
+        //             }
+        //         },
+        //         {
+        //             "kind": 0,
+        //             "name": "ETHPERP",
+        //             "expr": "\n(Market :name "ETHPERP"\n :tick-size 0.1\n :max-order-notional 1000000\n :max-taker-price-deviation 0.02\n :min-order-size 0.0001\n)",
+        //             "value": {
+        //                 "tickSize": "0.1",
+        //                 "minOrderSize": "0.0001",
+        //                 "maxOrderNotional": "1000000",
+        //                 "maxTakerPriceDeviation": "0.02"
+        //             }
+        //         }
+        //     ],
+        //         "timestamp": 1674260369,
+        //         "success": true
+        // }
+        let markets = response['value'];
+        markets = markets.filter ((market) => market['name'] !== 'DDXPERP');
+        return markets.map ((market) => {
+            const name = market['name'];
+            const base = name.slice (0, -4);
```

Comment:
is this just grabbing the part of the string which isn't 'PERP'?

Reply:
Yes that's correct